src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -2935,77 +3136,109 @@ private TAbstractAnalysisValue VisitInvocation_NonLambdaOrDelegateOrLocalFunctio
                 invokedAsDelegate: false, originalOperation: operation, defaultValue: value);
         }
 
-        private TAbstractAnalysisValue VisitInvocation_LambdaOrDelegateOrLocalFunction(
-            IInvocationOperation operation,
-            object? argument,
-            out HashSet<(IMethodSymbol method, IOperation? instance)>? resolvedMethodTargets)
+        private protected void MarkEscapedLambdasAndLocalFunctions(PointsToAbstractValue pointsToAbstractValue)
         {
-            var value = base.VisitInvocation(operation, argument);
+            Debug.Assert(IsPointsToAnalysis);
 
-            var knownTargetInvocations = false;
-            HashSet<(IMethodSymbol method, IOperation? instance)>? methodTargetsOptBuilder = null;
-            HashSet<IFlowAnonymousFunctionOperation>? lambdaTargets = null;
+            using var methodTargetsOptBuilder = PooledHashSet<(IMethodSymbol method, IOperation? instance)>.GetInstance();
+            using var lambdaTargets = PooledHashSet<IFlowAnonymousFunctionOperation>.GetInstance();
+            if (ResolveLambdaOrDelegateOrLocalFunctionTargets(pointsToAbstractValue, methodTargetsOptBuilder, lambdaTargets))
+            {
+                foreach (var (targetMethod, _) in methodTargetsOptBuilder)
+                {
+                    if (targetMethod.MethodKind == MethodKind.LocalFunction)
+                    {
+                        _escapedLocalFunctions.Add(targetMethod);
+                    }
+                }
 
-            if (HasPointsToAnalysisResult)
+                foreach (var flowAnonymousFunctionOperation in lambdaTargets)
+                {
+                    _escapedLambdas.Add(flowAnonymousFunctionOperation);
+                }
+            }
+        }
+
+        private bool ResolveLambdaOrDelegateOrLocalFunctionTargets(
+            PointsToAbstractValue invocationTarget,
+            PooledHashSet<(IMethodSymbol method, IOperation? instance)> methodTargetsOptBuilder,
+            PooledHashSet<IFlowAnonymousFunctionOperation> lambdaTargets)
+        => ResolveLambdaOrDelegateOrLocalFunctionTargetsCore(operation: null, invocationTarget, methodTargetsOptBuilder, lambdaTargets);
+
+        private bool ResolveLambdaOrDelegateOrLocalFunctionTargets(
+            IOperation operation,
+            PooledHashSet<(IMethodSymbol method, IOperation? instance)> methodTargetsOptBuilder,
+            PooledHashSet<IFlowAnonymousFunctionOperation> lambdaTargets)
+        => ResolveLambdaOrDelegateOrLocalFunctionTargetsCore(operation, invocationTarget: null, methodTargetsOptBuilder, lambdaTargets);
+
+        private bool ResolveLambdaOrDelegateOrLocalFunctionTargetsCore(
```

Comment:
This is almost entirely existing code, just refactored so we can use it both for context sensitive interprocedural analysis and computing escaped lambda/local functions targets.